### PR TITLE
Update java/build.gradle to not use deprecated features that were removed in gradle 8.0.

### PR DESCRIPTION
**Pull request by edgchen1**
### Description

Update java/build.gradle to not use deprecated features that were removed in gradle 8.0.
Also move gradle wrapper setup from a script into a step template.

### Motivation and Context

Fix builds which use hosted Mac agents and gradle.

Recently the system version of gradle got upgraded to 8.0. Even though we use an older gradle wrapper version, java/build.gradle is still processed with gradle 8.0 in the initial call to `gradle wrapper`.

### CODE REVIEW

**Comment by Craigacp** [comment]
Looks good, though if possible it might be good to move to a newer version of gradle in the wrapper. 6.8.3 isn't compatible with the latest versions of Java so it doesn't build there. The spotless plugin also needs to be bumped up to work on Java 17, that change is in #14027.

**Comment by edgchen1** [comment]
> Looks good, though if possible it might be good to move to a newer version of gradle in the wrapper. 6.8.3 isn't compatible with the latest versions of Java so it doesn't build there. The spotless plugin also needs to be bumped up to work on Java 17, that change is in #14027.

Yeah, I agree that it would be good to upgrade. Will leave that for another PR as I'm not sure how involved that is. Mainly want to fix the CI build failures here.

**Comment by Craigacp** [comment]
> Yeah, I agree that it would be good to upgrade. Will leave that for another PR as I'm not sure how involved that is. Mainly want to fix the CI build failures here.

For java it should be as simple as bumping the version number, but I don't know if there are othe constraints on the android build.

File: tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml [hunk 1]
```diff
@@ -235,8 +233,7 @@ stages:
 
       - template: "templates/use-android-ndk.yml"
 
-      - script: /bin/bash tools/ci_build/github/android/setup_gradle_wrapper.sh $(pwd)
-        displayName: Setup gradle wrapper to use gradle 6.8.3
+      - template: templates/set-up-gradle-wrapper-step.yml
```
Comment (pengwa) [inline tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml:236]: My understanding on the android related changes are just refactoring, am I right?

**Review by pengwa — APPROVED** [review]
Thanks for the fix!

File: tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml (hunk 1 above)
Comment (edgchen1) [reply tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml:236]: yes, build definition changes are refactoring. if we need to fix this urgently (before someone from onnxruntime-es takes a look), only the changes in java/build.gradle are necessary.

**Review by mszhanyi — APPROVED** [review]

**Review by stevenlix — APPROVED** [review]

**Pull request merged**